icicle/include/icicle/merkle/merkle_tree.h

```diff
@@ -116,17 +142,58 @@ namespace icicle {
      */
     eIcicleError verify(const MerkleProof& merkle_proof, bool& valid /*output*/) const
     {
-      // TODO: Implement Merkle path verification here
+      const auto& layer_hashes = m_backend->get_layer_hashes();
+      const uint nof_layers = layer_hashes.size();
+      auto [leaf, hash_input_size, leaf_idx] = merkle_proof.get_leaf();
+      HashConfig hash_config;
 
-      ICICLE_LOG_DEBUG << "In MerkleTree::verify()";
+      // offset in bytes to the selected leaf at the full tree
+      uint64_t leaf_element_start = leaf_idx * m_backend->get_leaf_element_size();
+
+      // Hash the leaves into hash_results
+      uint hash_output_size = layer_hashes[0].output_size();
```

Comment:
better use front() or at least assert it is not empty to avoid invalid memory access

Reply:
how front() helps?

Reply:
actually it doesn't. It is also undefined behavior if the vector is empty. 
The safe way seems to be https://en.cppreference.com/w/cpp/container/vector/at
or simply assert the vector size is large enough